include/deal.II/matrix_free/portable_matrix_free.h

```diff
@@ -871,6 +901,12 @@ namespace Portable
      */
     std::vector<unsigned int> row_start;
 
+    /**
+     * Mapping from color and index within the color to the cell index
+     * and level. For each color, stores the cell indices in order.
+     */
+    std::vector<std::vector<std::pair<unsigned int, unsigned int>>>
+      cell_level_index;
```

Comment:
why do you need this information? I would think this can be achieved using ``graph[color][index]->level()`` and ``graph[color][index]->index()``